plugins/aws/sts_provisioner.go

```diff
@@ -180,12 +182,18 @@ func (m CacheProviderFactory) NewAccessKeysProvider() aws.CredentialsProvider {
 
 // getAWSAuthConfigurationForProfile loads specified configurations from both config file and environment
 func getAWSAuthConfigurationForProfile(profile string) (*confighelpers.Config, error) {
+	// Disable log output produced by AWS Vault code
+	log.SetOutput(io.Discard)
+
 	// Read config file from the location set in AWS_CONFIG_FILE env var or from  ~/.aws/config
 	configFile, err := confighelpers.LoadConfigFromEnv()
 	if err != nil {
 		return nil, err
 	}
 
+	// Re-enable log output
+	log.SetOutput(os.Stderr)
+
```

Comment:
If we choose to go this way about silencing logs, i.e. wrapping some functionality in `log.SetOutput(io.Discard)` and `log.SetOutput(os.Stderr)`, I would rather have a nice generic wrapper function for that:

```
func ExecuteSilently[G interface{}, e error](functionalityToSilence func() (G, e)) (G, e) {
	// Disable log output
	log.SetOutput(io.Discard)
	// Re-enable log output
	defer log.SetOutput(os.Stderr)

	// Execute the functionality in a silent way
	return functionalityToSilence()
}
```

And then you could call that in our aws logic as such:

```
// Read config file from the location set in AWS_CONFIG_FILE env var or from  ~/.aws/config
	configFile, err := ExecuteSilently(confighelpers.LoadConfigFromEnv)
	if err != nil {
		return nil, err
	}
```

Give it a try!

Learn more about generics here: https://gobyexample.com/generics

Reply:
Thanks for that suggestion, I agree this is a better approach! 👍🏼 This is updated in https://github.com/1Password/shell-plugins/pull/297/commits/b179903cef133cc290d1283f3a21cc206013297d. Do you feel that `plugins/aws/utils.go` is a okay place for that generic function to live?